programs/utilities/spellbook.sc

```diff
@@ -0,0 +1,556 @@
+    'ci /spellbook <book> set <spell> </command>\n',
+    'm ie ', 'ci /spellbook cat_spells set "spawn cat" /summon cat\n',
+    'm You can add more spells to the book\n',
+    'ci /spellbook cat_spells set "spawn Tabby" /summon cat ~ ~ ~ {CatType:2}\n',
+    'm You can give yourself a copy of the spellbook with the spellbook give command\n',
+    'ci /spellbook cat_spells give\n',
+    'm Any changes you make to the spells will automatically update your spellbook when opened.\n'
+  ],
+  'shorthands' -> [
+    'pb \nSpellbook Shorthands\n',
+    'm The spellbook command has many shorthands to make life easier.\n',
+    'm \nFor example you can add a warp spell to your current location.\n',
+    'ci /spellbook <book> warp <spell>\n',
+    'm you can also be more specific\n',
+    'ci /spellbook transporter warp "Zero" at 0 0 0 in overworld\n',
+    'ci /spellbook <book> warp <spell> at <location> in <dimension>\n',
+    'm \nThere is also a player bot shorthand that creates two spells. ',
+    'm One to summon the player and the other kill the player\n',
+    'ci /spellbook farms bot "Which Hut" Dorothy\n',
+    'ci /spellbook <book> bot <spell> <botName> at 0 0 0 in <dimension>\n',
+    'ci /spellbook <book> bot <spell> <botName>\n',
+    'm \nSome farms don\'t require player based mob spawning. ',    
+    'm You also have the option to use force loading instead\n',
+    'm The forceload shorthand creates one spell to load the chunks and another to unload the chunks.\n',
+    'ci /spellbook farms forceload "The Iron Bucket" 32 32 64 64\n',
+    'ci /spellbook <book> forceload <spell> <from> <to>\n',
+    'ci /spellbook <book> forceload <spell> <from> <to> in <dimension>\n'
+  ],
+  'customize'-> [
+    'pb \nSpellbook Customization\n',
+    'm you can customize the color or the tooltips of spells with the following commands.\n',
+    'ci /spellbook <book> tooltip <spell> <customTooltip>\n',
+    'ci /spellbook <book> color <spell> <customColor>\n',
+  ],
+  'commands' -> [
+    'pb \nSpellbook Commands\n',
+    'ci /spellbook <book> set <spell> </command> ',
+    'm Add any command to the spellbook\n',
+    'ci /spellbook <book> give ',
+    'm Give the player a copy of the book\n',
+    'ci /spellbook <book> read ',
+    'm List the spells within a book\n',
+    'ci /spellbook list ',
+    'm List all of the spellbooks\n',
+    'ci /spellbook <book> delete ',
+    'm Delete the whole spellbook\n',
+    'ci /spellbook <book> remove <spell> ',
+    'm Delete a spell\n',
+    'ci /spellbook <book> tooltip <spell> <customTooltip> ',
+    'm Customize a spells tooltip\n',
+    'ci /spellbook <book> color <spell> <customColor>\n',
+    'm Customize a spells color\n',
+    'ci /spellbook <book> warp <spell>\n',
+    'm Create a warp spell at your location\n',
+    'ci /spellbook <book> bot <spell> <botName>\n',
+    'm Make a set of spells to summon and kill a player bot\n',
+    'ci /spellbook <book> forceload <spell> <from> <to>\n',
```

Comment:
Mostly that's the fist time I was really using format and I wasn't that familiar with it, then after I prioritized other feature.

Reply:
Ok... I tired that and I didn't like it. The differences between examples and references made it feel gross.
`/spellbook <book> set <spell> </command>` vs `/spellbook cat_spells set "spawn cat" /summon cat`
Return characters `\n` also posed a problem.
I did standardize them with a function. It might be slightly easier to add in the future now.

tldr: Its good enough as is. After trying it I decided not to add clipboard or suggestion support to the help pages.